zerver/views/auth.py

```diff
@@ -93,7 +95,7 @@ def login_or_register_remote_user(request, remote_username, user_profile, full_n
         # associated Zulip user account.
         if is_signup:
             # If they're trying to sign up, send them over to the PreregistrationUser flow.
-            return maybe_send_to_registration(request, remote_user_to_email(remote_username), full_name)
+            return maybe_send_to_registration(request, remote_user_to_email(remote_username), full_name, is_remote=True)
```

Comment:
I think this variable name should have be `password_required` with a default of `False`, rather than `is_remote` with a default of `True`.  That should give a much clearer idea of what it's for.